### ISSUE TITLE
Upgrade Guzzle to 6.0

### ISSUE THREAD

**Issue by scrothers** [authy/authy-php#29]
@vargasx @senekis @dcu @serargz 

There are open issues and open PRs for a composer.json change.

Does this library need to be forked or are you guys going to actually respond to these PRs and issues?

**Comment by merlingore** [comment]
I think the repo with actual changes was deleted so I've [resubmitted it](https://github.com/authy/authy-php/pull/30). Hopefully that one will be responded to!

**Comment by scrothers** [comment]
These issues have been open for over a half a year in most cases. It's really inexcusable that they have had no response. Authy obviously is not a fan of developers.

**Comment by merlingore** [comment]
Seems like they've merged some recent requests, since the old one didn't have a repo attached (the repo owner inadvertently deleted it) that might have been the cause of the delay. Let's hope they merge this one.

**Comment by zhilevan** [comment]
I have similar problem, when I want to use this package  on drupal 8 faced with

```
composer require authy/php
Using version ^2.5 for authy/php
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Can only install one of: guzzlehttp/guzzle[5.3.x-dev, 6.2.1].
    - Can only install one of: guzzlehttp/guzzle[6.2.1, 5.3.x-dev].
    - Can only install one of: guzzlehttp/guzzle[5.3.x-dev, 6.2.1].
    - authy/php v2.5 requires guzzlehttp/guzzle ~5.0 -> satisfiable by guzzlehttp/guzzle[5.3.x-dev].
    - Installation request for authy/php ^2.5 -> satisfiable by authy/php[v2.5].
    - Installation request for guzzlehttp/guzzle (locked at 6.2.1, required as ^6.2.1) -> satisfiable by guzzlehttp/guzzle[6.2.1].


Installation failed, reverting ./composer.json to its original content.

```

**Comment by mykevin81** [comment]
Me too, I need to include oauth2.0 package into it and it's requires guzzle 6

**Comment by serargz** [comment]
hi @scrothers, sorry for the ridiculously long time to respond here. We're catching up with issues and PRs. The PR https://github.com/authy/authy-php/pull/40 was merged (which includes support for Guzzle 6) and some other improvements will be published in a release in the next few days.

**Comment by scrothers** [comment]
@serargz any chance we're gonna see unit tests that work too? The entire reason why my organization wont implement Authy is because of the lack of support from Authy on official libraries. Right now I see 3 failures in unit tests, are you now in charge of managing this library?

**Comment by serargz** [comment]
@scrothers yes, absolutely. The PR https://github.com/authy/authy-php/pull/40 is under review with those changes. I'm sorry, we had too many things in our plate that prevented us from maintaining our libraries properly, but we're working to keep them up to date from now.

**Comment by serargz** [comment]
@scrothers we have released v3.0 which includes Guzzle 6.x and tests were fixed too. The next release will include support for OneTouch.
